victron_ble/devices/solar_charger.py

```diff
@@ -0,0 +1,90 @@
+from enum import Enum
+from typing import Any, Dict
+
+from construct import Int16sl, Int16ul, Struct
+
+from victron_ble.devices.base import Device, DeviceData
+
+
+class ChargeState(Enum):
+    OFF = 0
+    BULK = 3
+    ABSORPTION = 4
+    FLOAT = 5
+
+
+class SolarChargerData(DeviceData):
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self._data = data
+
+    def get_charge_state(self) -> ChargeState:
+        """
+        Return an enum indicating the current charging state
+        """
+        return self._data["charge_state"]
+
+    def get_battery_voltage(self) -> float:
+        """
+        Return the battery voltage in volts
+        """
+        return self._data["battery_voltage"]
+
+    def get_battery_charging_current(self) -> float:
+        """
+        Return the battery charging current in amps
+        """
+        return self._data["battery_charging_current"]
+
+    def get_yield_today(self) -> float:
+        """
+        Return the yield_today in Wh
+        """
+        return self._data["yield_today"]
+
+    def get_solar_power(self) -> float:
+        """
+        Return the current solar power in W
+        """
+        return self._data["solar_power"]
+
+    def get_external_device_load(self) -> float:
+        """
+        Return the external device load in amps
+        """
+        return self._data["external_device_load"]
+
+
+class SolarCharger(Device):
+
+    PACKET = Struct(
+        # Charge State:   0 - Off
+        #                 3 - Bulk
+        #                 4 - Absorption
+        #                 5 - Float
+        "charge_state" / Int16sl,
+        # Battery voltage reading in 0.01V increments
+        "battery_voltage" / Int16ul,
+        # Battery charging Current reading in 0.1A increments
+        "battery_charging_current" / Int16ul,
+        # Todays solar power yield in 10Wh increments
+        "yield_today" / Int16ul,
+        # Current power from solar in 1W increments
+        "solar_power" / Int16ul,
+        # External device load in 0.1A increments
+        "external_device_load" / Int16ul,
```

Comment:
How sure are you about this one?

I see this right now:

```
  "payload": {
    "battery_charging_current": 0.0,
    "battery_voltage": 25.3,
    "charge_state": "off",
    "external_device_load": 51.1,
    "solar_power": 0,
    "yield_today": 1420
  }
```

Not sure that external load makes any sense when it's not making power. Also, this number isn't visible anywhere in the UI.

Reply:
@stefanor Which model are you testing with? I think only the smaller output models have dedicated load outputs so I wonder if these values are nonsense for the other models.

Reply:
I had to strip off the upper bytes to get the load:
`            pkt.external_device_load & 0x01FF
`
Maybe the upper bytes indicate if the load-output exists or not? 
Could you output the following for debug purposes:            
`pkt.external_device_load & 0xFE00
`
Beside this:
How many values do you see in the Victron App for your Solar Charger? 
Any additional data?

Not sure if this is part of the advertising data or if Victron has hardcoded this on a device level?!

Reply:
> Which model are you testing with?

I have an MPPT 100/50

> Could you output the following for debug purposes:

external_device_load & 0xFE00: fe00

>  How many values do you see in the Victron App for your Solar Charger?

For instant data:

![image](https://user-images.githubusercontent.com/442117/210159311-570c3031-986d-44a7-921b-804fe9451306.png)

I haven't been able to make sense of any values in the last 6 bytes of the announcement packet

Reply:
So it looks like there is no way based on the data to see if there is a load output or not.
Should we hardcode this based on the model type?

Reply:
For me the raw value is always `0xffff`, so I suggest skipping it if it is `0xffff`

Reply:
@stefanor: 
I've put a check to the code for this. Do you mind sending me some advertising data plus the matching key, so that we can add this to the test?

Reply:
```
    def test_parse_mppt100(self) -> None:
        data = "100249a0013a399bbb3e36d7237c7687f96e45dc"
        actual = SolarCharger("9b3509d3d7aba706846214ca64500d0c").parse(
            bytes.fromhex(data)
        )
        assert actual.get_battery_charging_current() == 10.1
        assert actual.get_battery_voltage() == 25.55
        assert actual.get_charge_state() == OperationMode.BULK
        assert actual.get_solar_power() == 265
        assert actual.get_yield_today() == 500
        assert actual.get_external_device_load() == None
        assert actual.get_model_name() == "BlueSolar MPPT 100|50 rev2"
```

Reply:
Thanks...added to the test...